diff --git a/app/Http/Kernel.php b/app/Http/Kernel.php
index 2741c0a..ed5a054 100644
--- a/app/Http/Kernel.php
+++ b/app/Http/Kernel.php
@@ -61,6 +61,7 @@ class Kernel extends HttpKernel
         'signed' => \Illuminate\Routing\Middleware\ValidateSignature::class,
         'throttle' => \Illuminate\Routing\Middleware\ThrottleRequests::class,
         'verified' => \Illuminate\Auth\Middleware\EnsureEmailIsVerified::class,
+        'hasPermission' => \App\Http\Middleware\HasPermission::class
     ];
 
     /**
diff --git a/app/Http/Middleware/HasPermission.php b/app/Http/Middleware/HasPermission.php
new file mode 100644
index 0000000..a39f4d8
--- /dev/null
+++ b/app/Http/Middleware/HasPermission.php
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Support\Facades\Gate;
+
+class HasPermission
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $permissionName)
+    {
+        if(Gate::denies($permissionName)) {
+            throw new AuthorizationException();
+        }
+
+        return $next($request);
+    }
+}
diff --git a/routes/web.php b/routes/web.php
index 6975bb0..0777bba 100644
--- a/routes/web.php
+++ b/routes/web.php
@@ -33,7 +33,7 @@ Route::group(["middleware" => ['auth']], function () {
     Route::post('/documentation/{patientId}', 'DocumentationController@store')->name('documentation');
     Route::get('/documentations', 'DocumentationController@index')->name('documentations');
 
-    Route::get('/patients', 'PatientController@index')->name('patients');
+    Route::get('/patients', 'PatientController@index')->name('patients')->middleware('hasPermission:admin-patient');
     Route::get('/patient/{id}', 'PatientController@edit')->name('patient');
     Route::get('/patient/', 'PatientController@create')->name('newpatient');
     Route::post('/patient/', 'PatientController@store')->name('newpatient');
